app/services/CollectionService.scala

```diff
@@ -97,15 +97,20 @@ object CollectionService {
       config: ConfigJson,
       containerService: ContainerService
   ): Option[StoriesVisibleByStage] = {
-    val cConfigJson = config.collections.get(collectionId).get
     val stages =
       CollectionService.getStoriesForCollectionStages(collectionId, collection)
     config.collections.get(collectionId).flatMap(_.`type`) match {
       case Some(cType) =>
+        val cConfigJson = config.collections(collectionId)
```

Comment:
This will through an exception if the collection is not present, and is exactly equivalent to the `config.collections.get(collectionId).get` you had before.

Instead, you could do `val cConfigJson = config.collections.get(collectionId)` and drop the `Some(...)` that you have wrapped `cConfigJson` in below.

Reply:
I do prefer the look of that suggestion, but just to check my understanding - wouldn't this code only get executed after line 102, which should guarantee the collection is present at that point?

Reply:
That is an excellent point. I missed that you already know that the collection exists. But much like me, the complier isn't clever enough to detect that your actually accessing it in a safe place way.

Looking at this again, you could refactor:
```
    for {
      cConfigJson <- config.collections.get(collectionId)
      cType <- cConfigJson.`type`
    } yield {
        StoriesVisibleByStage(
          containerService
            .getStoriesVisible(cType, stages._1, Some(cConfigJson)),
          containerService.getStoriesVisible(
            cType,
            stages._2,
            Some(cConfigJson)
          )
        )
    }
```

Reply:
Thanks, that does look a bit more concise and seems to work!